addons/Connection/src/presenter.js

```diff
@@ -1017,6 +1031,94 @@ function AddonConnection_create() {
+        }, userAnswers);
+        correctAnswers.forEach(function(element, index) {
+            this[index] = element.join('');
+        }, correctAnswers);
+
+        if (userAnswers == null || correctAnswers == null) return false;
+        if (userAnswers.length !== correctAnswers.length) return false;
+      
+        userAnswers.sort();
+        correctAnswers.sort();
+      
+        for (var i = 0; i < userAnswers.length; ++i) {
+          if (userAnswers[i] !== correctAnswers[i]) return false;
+        }
+        return true;
+      }
+
+    function appendAnswerDiv (column, elementId, answerDiv) {
+        var element = $('<table class="answerItem" id="connection-' + elementId + '"></div>');
+        var row = $('<tr></tr>');
+        var innerElement = $('<td class="inner"></td>');
+        var newRow = $('<tr></tr>');
+        var newCell = $('<td class="connectionItemWrapper"></td>');
+        innerElement.append(answerDiv);
+        row.append(innerElement);
+        element.append(row);
+        newCell.append(element);
+        newRow.append(newCell);
+        column.append(newRow);
+    }
+
+    presenter.addAnswersElements = function (view, model, correctAnswers) {
+        // left side
+        userAnswers = []
+        if (presenter.printableState.id.length > 0) {
+            for (var i = 0; i < presenter.printableState.id.length; i++) {
+                var pair = presenter.printableState.id[i].split(':');
+                userAnswers.push(pair);
+            }
+        }
+
+        var leftAnswerColumn = $(view).find('.answersLeftColumn:first').find('.content:first');
+        var rightAnswerColumn = $(view).find('.answersRightColumn:first').find('.content:first');
+        
+        // Left answer column
+        for (var i = 0; i < model["Left column"].length; i++) {
+            var element = presenter.elements[i];
+            var leftId = element.id;
+            var userConnectionsWithLeftId = userAnswers.filter((userAnswer) => {
+                return(userAnswer[0] == leftId);
+            })
+            if (userConnectionsWithLeftId.length == 0) {
+                var emptyAnswer = [`${leftId}`, ''];
+                userConnectionsWithLeftId.push(emptyAnswer);
+            }
+            var correctConnectionsWithLeftId = correctAnswers.filter((correctAnswer) => {
+                return(correctAnswer[0] == leftId);
+            })
+            if ( isAnswersEqualToCorrectResults(userConnectionsWithLeftId, correctConnectionsWithLeftId) ) {
+                var correctAnswerDiv = $('<div class="correctAnswerDiv"></div>');
+                appendAnswerDiv(leftAnswerColumn, leftId, correctAnswerDiv);
+            } else {
+                var inCorrectAnswerDiv = $('<div class="inCorrectAnswerDiv"></div>');
+                appendAnswerDiv(leftAnswerColumn, leftId, inCorrectAnswerDiv);
+            }
+        }
+        // Right answer column
+        for (var i = 0; i < model["Right column"].length; i++) {
```

Comment:
Kodzik wygląda bardzo podobnie to tego dla right column - sugeruję ubrać to w funkcję.

Reply:
Podobny, ale logika jest na tyle różna, że moim zdaniem próba wyciągnięcia tego kodu do oddzielnej funkcji bardziej by namieszała niż ułatwiła.